json-fleece-openapi3/src/Fleece/OpenApi3.hs

```diff
@@ -241,24 +242,24 @@ lookupRequestBody operationKey operation =
       pure Nothing
 
 data SchemaTypeInfoWithDeps = SchemaTypeInfoWithDeps
-  { schemaTypeInfoDependent :: CGU.SchemaTypeInfo
+  { schemaTypeInfoDependent :: CGU.SchemaTypeInfoOrRef
```

Comment:
I agree there could probably be some further simplification in the abstractions here around `SchemaTypeInfoWithDeps` (actually `SchemaTypeInfoOrRef` was one initial attempt in that direction). The tricky thing is we have refs which need to be resolved later (at code generation time, once we have all the top level names, since refs under components could refer to other refs in components), but we also have a lot of code (especially the operations section) that directly produces `SchemaTypeInfo` (ex: `lookupRequestBodySchema`). I think it might be feasible to move all the `SchemaTypeInfo` resolution into `generateCodeGenDataFormat` by expanding `resolveRefTypeInfo` to work on more than just refs for instance, but this would be a pretty significant change that I'd want to discuss through first. I also think it may be reasonable to wait and do this refactoring at a later stage, as implementing some of the unsupported features (like nested inline objects) might change how we approach the abstractions.